### PR TITLE
feat: add S3 import functionality which is supported for MySQL instances

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- add S3 import functionality which is supported for MySQL instances
- add new example for demonstrating and verifying S3 import functionality
- update pre-commit hook dependencies & update required provider syntax
- `snapshot_identifier` and `replicate_source_db` defaults changed to `null` due to error `"s3_import": conflicts with snapshot_identifier|replicate_source_db`
- add `sensitive = true` to password outputs due to error message
- gitignore updated using GitHub provided default standard + 0.14 lockfile ignore

## Motivation and Context
- closes #247 
- closes #258 

## Breaking Changes
- No

## How Has This Been Tested?
- Tested using new example directory `s3-import-mysql`; directions added for others to replicate testing/confirmation

### CODE REVIEW

File: examples/s3-import-mysql/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+CREATE USER 's3_import_user'@'localhost' IDENTIFIED BY 'YourPwdShouldBeLongAndSecure!';
+GRANT ALL PRIVILEGES ON * . * TO 's3_import_user'@'localhost';
+FLUSH PRIVILEGES;
+```
+
+3. Use Percona Xtrabackup container to dump database and upload to S3
+
+```bash
+$ docker run --name percona-xtrabackup-8.0 --mount type=bind,src=/tmp/backup,dst=/backup --volumes-from percona-server-mysql-8.0.20 percona/percona-xtrabackup:8.0 xtrabackup --backup --data-dir=/var/lib/mysql --target-dir=/backup --user=root --password=root
+$ s3 sync /tmp/backup/ s3://s3-import-<UPDATE-NAME>/
```
Comment (antonbabenko) [inline examples/s3-import-mysql/README.md:40]: Can you please make a tiny SQL backup file in this example dir, create S3 bucket and upload it to the new bucket using [aws_s3_bucket_object](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_object) resource, and then reference it in a test?

The goal is to run just "terraform apply" to get end-to-end examples to work.

Comment (bryantbiggs) [reply examples/s3-import-mysql/README.md:40]: yep - I can add that

**Comment by bryantbiggs** [comment]
@antonbabenko this one should be ready for review - I'll take a look at the one you linked and see how best to proceed with that one

File: examples/s3-import-mysql/README.md (hunk 1 above)
Comment (bryantbiggs) [reply examples/s3-import-mysql/README.md]: FYI - there are quite a few files (you can see them in https://github.com/terraform-aws-modules/terraform-aws-rds/pull/289/commits/6be766667530a18d402fbc99efbc486b122bfe79) so I couldn't use the s3 bucket object resource. I did try with the `fileset()` but that doesn't support recursively adding everything so I had to go with a local-exec to make it work

Comment (antonbabenko) [reply examples/s3-import-mysql/README.md]: I see, I didn't know there would be so many files needed for the small demo. Thought there would be just 1-3 :)

Anyway, your solution with `unzip and s3 sync` looks perfect as the working example.

**Comment by antonbabenko** [comment]
[v2.21.0](https://github.com/terraform-aws-modules/terraform-aws-rds/releases/v2.21.0) has been just released.

**Pull request merged**